tests/src/test/java/alluxio/underfs/sleepfs/SleepingUnderFileSystem.java

```diff
@@ -0,0 +1,250 @@
+  }
+
+  @Override
+  public String getOwner(String path) throws IOException {
+    sleepIfNecessary(mOptions.getGetOwnerMs());
+    return super.getOwner(cleanPath(path));
+  }
+
+  @Override
+  public long getSpace(String path, SpaceType type) throws IOException {
+    sleepIfNecessary(mOptions.getGetSpaceMs());
+    return super.getSpace(cleanPath(path), type);
+  }
+
+  @Override
+  public String getUnderFSType() {
+    sleepIfNecessary(mOptions.getGetUnderFSTypeMs());
+    return super.getUnderFSType();
+  }
+
+  @Override
+  public boolean isDirectory(String path) throws IOException {
+    sleepIfNecessary(mOptions.getIsDirectoryMs());
+    return super.isDirectory(cleanPath(path));
+  }
+
+  @Override
+  public boolean isFile(String path) throws IOException {
+    sleepIfNecessary(mOptions.getIsFileMs());
+    return super.isFile(cleanPath(path));
+  }
+
+  @Override
+  public UnderFileStatus[] listStatus(String path) throws IOException {
+    sleepIfNecessary(mOptions.getListStatusMs());
+    return super.listStatus(cleanPath(path));
+  }
+
+  @Override
+  public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
+    sleepIfNecessary(mOptions.getMkdirsMs());
+    return super.mkdirs(cleanPath(path), options);
+  }
+
+  @Override
+  public InputStream open(String path, OpenOptions options) throws IOException {
+    sleepIfNecessary(mOptions.getOpenMs());
+    return super.open(cleanPath(path), options);
+  }
+
+  @Override
+  public boolean renameDirectory(String src, String dst) throws IOException {
+    sleepIfNecessary(mOptions.getRenameDirectoryMs());
+    return super.renameDirectory(cleanPath(src), cleanPath(dst));
+  }
+
+  @Override
+  public boolean renameFile(String src, String dst) throws IOException {
+    if (!PathUtils.isTemporaryFileName(src)) {
+      sleepIfNecessary(mOptions.getRenameFileMs());
+    } else {
+      sleepIfNecessary(mOptions.getRenameTemporaryFileMs());
+    }
+    return super.renameFile(cleanPath(src), cleanPath(dst));
+  }
+
+  @Override
+  public void setConf(Object conf) {
+    sleepIfNecessary(mOptions.getSetConfMs());
+    super.setConf(conf);
+  }
+
+  @Override
+  public void setOwner(String path, String owner, String group) throws IOException {
+    sleepIfNecessary(mOptions.getSetOwnerMs());
+    super.setOwner(cleanPath(path), owner, group);
+  }
+
+  @Override
+  public void setMode(String path, short mode) throws IOException {
+    sleepIfNecessary(mOptions.getSetModeMs());
+    super.setMode(cleanPath(path), mode);
+  }
+
+  @Override
+  public boolean supportsFlush() {
+    sleepIfNecessary(mOptions.getSupportsFlushMs());
+    return super.supportsFlush();
+  }
+
+  /**
+   * Removes the sleep scheme from the path if it exists.
+   *
+   * @param path the path to clean
+   * @return a path without the sleep scheme
+   */
+  private String cleanPath(String path) {
+    if (path.startsWith("sleep://")) {
+      return path.substring(8);
```

Comment:
it would be less error prone the parse the path as AlluxioURI and return the path portion